src/callbacks/sync_selected.cpp

```diff
@@ -153,13 +169,14 @@ void sync_selected::init_cuda_profiler(
 }
 
 void sync_selected::setup(model *m) {
+  init_cuda_profiler(*(m->get_comm()));
```

Comment:
This changes the behavior to start nvprof during the callback setup, instead of at construction. This may cause the profiler to miss out on setup in other objects, like layers. @naoyam @ndryden Is this behavior change OK?

Reply:
If this is the wrong fix, that's fine. But this needs to be fixed somehow. There is no `comm` available during factory time (and it will be a really hard sell to convince me that this one outlier is worth the baggage) and there's a `barrier` in there so it seems to matter for this. `setup` is the first time a callback sees a `comm` (via the `model`), so it seemed a logical place. If NVPROF should be initialized globally, perhaps moving this up the stack is the best answer. Let me know.